cpp/integ/date_time_test.cc

```diff
@@ -0,0 +1,137 @@
+// Copyright (C) 2017-2018 Dremio Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <time.h>
+#include <math.h>
+#include <gtest/gtest.h>
+#include "arrow/memory_pool.h"
+#include "integ/test_util.h"
+#include "gandiva/projector.h"
+#include "gandiva/tree_expr_builder.h"
+
+namespace gandiva {
+
+using arrow::int32;
+using arrow::int64;
+using arrow::float32;
+using arrow::boolean;
+using arrow::date64;
+
+class TestProjector : public ::testing::Test {
+ public:
+  void SetUp() { pool_ = arrow::default_memory_pool(); }
+
+ protected:
+  arrow::MemoryPool* pool_;
+};
+
+long long MillisSince(time_t base_line, int yy, int mm, int dd, int hr, int min, int sec) {
+  struct tm given_ts = {0};
+  given_ts.tm_year = (yy - 1900);
+  given_ts.tm_mon = (mm - 1);
+  given_ts.tm_mday = dd;
+  given_ts.tm_hour = hr;
+  given_ts.tm_min = min;
+  given_ts.tm_sec = sec;
+
+  return (lround(difftime(mktime(&given_ts), base_line)) * 1000);
+}
+
+TEST_F(TestProjector, TestTime) {
+  auto field0 = field("f0", date64());
+  auto field1 = field("f1", time32(arrow::TimeUnit::MILLI));
+  auto field2 = field("f2", timestamp(arrow::TimeUnit::MILLI));
+  auto schema = arrow::schema({field0, field2});
+
+  // output fields
+  auto field_year = field("yy", int64());
+  auto field_month = field("mm", int64());
+  auto field_day = field("dd", int64());
+  auto field_hour = field("hh", int64());
+
+  // extract year and month from date
+  auto date2year_expr = TreeExprBuilder::MakeExpression("extractYear", {field0}, field_year);
+  auto date2month_expr = TreeExprBuilder::MakeExpression("extractMonth", {field0}, field_month);
+
+  // extract day and hour from time32
+  auto time2day_expr = TreeExprBuilder::MakeExpression("extractDay", {field1}, field_day);
+  auto time2hour_expr = TreeExprBuilder::MakeExpression("extractHour", {field1}, field_hour);
+
+  // extract month and day from timestamp
+  auto ts2month_expr = TreeExprBuilder::MakeExpression("extractMonth", {field2}, field_month);
+  auto ts2day_expr = TreeExprBuilder::MakeExpression("extractDay", {field2}, field_day);
+
+  std::shared_ptr<Projector> projector;
+  Status status = Projector::Make(schema, {date2year_expr, date2month_expr, ts2month_expr, ts2day_expr}, pool_, &projector);
```

Comment:
looks like this breaks our formatting rules..
i think integ tests are not covered under lint - can you please fix (or) create an issue to include..